### DISCUSSION TITLE
How to loop through filename using "Upload File" block

### DISCUSSION THREAD

**Discussion by JoelSund** [AutomaApp/automa#1206] [Q&A] [answered]
First, thank you for this marvelous tool !

I need to upload many files.

I have a list with file names.
cat file_list.txt
c:\\tmp\\file1.txt
c:\\tmp\\file2.txt

I need to input those values in "Upload File" block so it will upload a new file (from list) on each time it runs
<img width="1430" alt="11" src="https://user-images.githubusercontent.com/6420434/225226620-94d4c22c-a93f-4924-991b-6496387fb25b.png">

<img width="1196" alt="12" src="https://user-images.githubusercontent.com/6420434/225110814-6519396c-dc5e-4edf-95a4-b2c5f43d97d7.png">
.

**Comment by JoelSund** [comment]
So, is it possible to assign "upload file" block a variable ($$filename), et getting $$filename iteratively from list ?

**Comment by JoelSund** [comment, marked as answer]
Resolved.


File 
<img width="704" alt="13" src="https://user-images.githubusercontent.com/6420434/225229537-e8e113d1-d1c9-4a93-8dd2-f2a56d1da954.png">
Must be .json ,    formated like ["C:\\dir\\question_0.pdf","C:\\dir\\question_1.pdf","C:\\dir\\question_2.pdf"] 


And the "Upload file" Block must contain {{loopData@BDThw0}}  where BDThw0 is the ID of Loop data.

**Comment by iheroformyself** [comment]
how to upload in google form ?

**Comment by CalvinLKT** [comment]
May I know how to upload a PDF one by one?
